src/lib/core/ErrorStr.h

```diff
@@ -49,6 +49,7 @@ struct ErrorFormatter
 };
 
 extern const char * ErrorStr(CHIP_ERROR err, bool withSourceLocation = true);
+extern void ErrorStr(CHIP_ERROR err, bool withSourceLocation, char * buf, uint16_t bufSize);
```

Comment:
How about using `MutableByteSpan` instead of pointer and size? Also for convenience, maybe this function should return `const char *` just like the other one?

Reply:
Will change it to return const char*, but using a span here creates circualr dependancy.